### PR TITLE
For https://github.com/Symas/LogAnalyzer/issues/4, add support for py…

### PR DESCRIPTION
**Pull request by quanah**
…thon3

### CODE REVIEW

**Comment by quanah** [comment]
With python2.7:
```
quanah@ub16:~/git/symas/LogAnalyzer$ python2.7 ./symasLog.py ~/customers/dnb/idaas/symas/log/slapd_log.log
('Processing file : ', '/home/quanah/customers/dnb/idaas/symas/log/slapd_log.log')
.........Done, reading    81410 lines


==========================================
== LDAP server restart #    0          ==
==========================================
Total Log Lines Analyzed: 81410

----------- Access Log Output ------------

Start of Logs:     2016-11-04 08:44:57
End of Logs:       2016-11-04 09:56:40

Processed Log Time: 4303 seconds (1:11:43)

----------- Global statistics ------------
Total Connections:                 294
 - StartTLS Connections:             5
 - LDAPI Connections:         ***TODO***
Peak Concurrent Connections:  ***TODO***
Total Operations:                   31
Total Results:                      12
Total Returned entries:              6
Overall Performance:          0.01 operations/sec, 0.00 entries/sec

--------------- Operations ---------------
Abandons:                            4 (0.00/sec) (0.01/session)
Binds:                               7 (0.00/sec) (0.02/session)
    Simple:                          2 (0.00/sec) (0.01/session)
    SASL:                            5 (0.00/sec) (0.02/session)
Extendeds:                           5 (0.00/sec) (0.02/session)
Searches:                           15 (0.00/sec) (0.05/session)

('Number of Opened sessions : ', 294)
('Number of Closed sessions : ', 290)
('Number of Zombie sessions : ', 5)


-------------- Errors found --------------

------------- Warnings found -------------
Not indexed attributes :


------- Miscellaneous informations -------
Average searches per connection :     0.05
('Number of returned entries: ', 6)
Average entries per search :     0.40
('Number of entries returned, grouped : ', 12)
('Average number of searches per second when some searches are done :', 1)
('filters : ', 0)
---------------- Restarts ----------------
```

**Comment by quanah** [comment]
With python 3:
```
quanah@ub16:~/git/symas/LogAnalyzer$ python3 ./symasLog.py ~/customers/dnb/idaas/symas/log/slapd_log.log
Processing file :  /home/quanah/customers/dnb/idaas/symas/log/slapd_log.log
.........Done, reading    81410 lines


==========================================
== LDAP server restart #    0          ==
==========================================
Total Log Lines Analyzed: 81410
----------- Access Log Output ------------
Start of Logs:     2016-11-04 08:44:57
End of Logs:       2016-11-04 09:56:40
Processed Log Time: 4303 seconds (1:11:43)
----------- Global statistics ------------
Total Connections:                 294
 - StartTLS Connections:             5
 - LDAPI Connections:         ***TODO***
Peak Concurrent Connections:  ***TODO***
Total Operations:                   31
Total Results:                      12
Total Returned entries:              6
Overall Performance:          0.01 operations/sec, 0.00 entries/sec
--------------- Operations ---------------
Abandons:                            4 (0.00/sec) (0.01/session)
Binds:                               7 (0.00/sec) (0.02/session)
    Simple:                          2 (0.00/sec) (0.01/session)
    SASL:                            5 (0.00/sec) (0.02/session)
Extendeds:                           5 (0.00/sec) (0.02/session)
Searches:                           15 (0.00/sec) (0.05/session)
Number of Opened sessions :  294
Number of Closed sessions :  290
Number of Zombie sessions :  5


-------------- Errors found --------------

------------- Warnings found -------------
Not indexed attributes :


------- Miscellaneous informations -------
Average searches per connection :     0.05
Number of returned entries:  6
Average entries per search :     0.40
Number of entries returned, grouped :  12
Average number of searches per second when some searches are done : 1.875
filters :  0
---------------- Restarts ----------------
```

**Comment by quanah** [comment]
Hm, looks like I need to do a little formatting fixing for 2.6. ;)

**Comment by mistotebe** [comment]
The bit with "Processing file:" can be fixed with a `from __future__ import print_function` (should work from 2.6 onwards).

File: symasLog.py [hunk 1]
```diff
@@ -495,95 +495,107 @@ def printResult( counters, restarts, args ) :
...
-        print ""
-        print "-------------- Errors found --------------"
+        print ("")
+        print ("-------------- Errors found --------------")
     
         errors = [0]*81
     
         for result in counters.ldapResult:
             if ( result != None ) :
-                for sessionId in result.iterkeys():
```
Comment (mistotebe) [inline symasLog.py]: If you change this to `for key in result:`, you don't have to check for the python version.

File: symasLog.py [hunk 2]
```diff
@@ -612,21 +624,25 @@ def printResult( counters, restarts, args ) :
...
                         result[sessionId].prettyPrint( "    " )
     
-        print ""
-        print "------------- Warnings found -------------"
-        print "Not indexed attributes : "
+        print ("")
+        print ("------------- Warnings found -------------")
+        print ("Not indexed attributes : ")
         
-        for aKey in counters.notIndexedAttributes.iterkeys():
```
Comment (mistotebe) [inline symasLog.py]: Same here.

File: utils/symasLogUtils.py [hunk 3]
```diff
@@ -338,15 +338,15 @@ def processNotIndexedAttribute( line, notIndexedAttributes ):
 #-------------------------------------------------------------
 def printNonZero( text, value ):
     if ( value != 0 ):
-        print ( text + "{0:8}" ).format( value )
+        print ( text + "{0:8}".format( value ))
```
Comment (mistotebe) [inline utils/symasLogUtils.py]: :+1:

**Comment by quanah** [comment]
Fixed the iterkeys bits, and reran on testing input log.  Looks good, going to commit this in.

**Pull request merged**